src/lib/y2network/ipoib_mode.rb

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Network
+  # This class represents the supported IPoIB transport modes.
+  class IpoibMode
+    class << self
+      # Returns all the existing modes
+      #
+      # @return [Array<IpoibMode>]
+      def all
+        @all ||= IpoibMode.constants
```

Comment:
Not for this PR, but we might think about putting this code in a common place (a mixin or something like that).